File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    
    sizes = {}
    
    for t in tangerine:
        if t not in sizes:
            sizes[t] = 0
        sizes[t] += 1
    
    kv = list(sizes.items())
    kv.sort(key=lambda x: x[1])
    
    answer = 0
    while k > 0:
        _, val = kv.pop()
        answer += 1
        k -= val
    
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.swift
import Foundation

func solution(_ k:Int, _ tangerine:[Int]) -> Int {
    
    var dict: [Int: Int] = [:]
    
    for t in tangerine {
        dict[t, default: 0] += 1
    }
    
    var kv: [(key: Int, cnt: Int)] = []
    
    for (key, cnt) in dict {
        kv.append((key, cnt))
    }
    
    kv.sort { $0.cnt < $1.cnt }
    
    var k = k
    var answer = 0
    
    while k > 0 {
        let (key, cnt) = kv.removeLast()
        answer += 1
        k -= cnt
    }
    
    return answer
}